### DISCUSSION TITLE
Seq won't display in Safari but works in Chrome on Apple M2 Sonoma

### DISCUSSION THREAD

**Discussion by Mathiasw** [datalust/seq-tickets#1990] [General]
I am running Seq in Docker desktop on a Apple M2 Sonoma.

When I try to login with Safari (17.0 - the latest) I simply get bounced back to the logon screen like this:

<img width="1272" alt="Screenshot 2023-10-17 at 08 11 54" src="https://github.com/datalust/seq-tickets/assets/733067/e8f559fa-b6d7-4838-9cf2-3b92b12350b1">

Also, every login attempt results in the URL in the browser getting longer, after two attempts it's: http://localhost:8020/#/login?to=%2Flogin%3Fto%3D%252Flogin%253Fto%253D%25252Flogin%25253Fto%25253D%2525252Flogin

Using Chrome as the browser works perfectly:

<img width="1374" alt="Screenshot 2023-10-17 at 08 14 55" src="https://github.com/datalust/seq-tickets/assets/733067/5994f631-650b-4b97-bef1-d5a8e364cd62">

I'm happy to continue to just use Chrome, but I thought you might be interested.

**Comment by nblumhardt** [comment]
Interesting - thanks for letting us know, we'll take a look 👍

**Reply by Mathiasw** [reply]
Thanks, if you need any other debugging information or need a fix testing, just let me know.

**Reply by nblumhardt** [reply]
Thanks, we'll have an update on this one shortly.

**Reply by nblumhardt** [reply]
Hi Mathias, thanks for your patience.

We've tried but not successfully replicated this; would it be possible to check what's going on in your:

a) browser developer tools console and
b) browser developer tools "network" tab

when the redirect back to the login page happens? You may need to adjust the developer tools settings to preserve the console and network tab contents between redirects, depending on what's going on.

If you can't spot anything obvious, sharing some (redacted :-)) details with us via our `support@datalust.co` address should help us pin it down.

Best regards,
Nick

**Reply by Mathiasw** [reply]
I will do that now.
I also tested om iPadPro, and I get the same issue with Safari, Firefox and Chrome on the iPad. Every login attempt just redirects you back, and the URL in the browser gets long with each attempt until it starts to look like this:

http://localhost:8020/#/login?to=%2Flogin%3Fto%3D%252Flogin%253Fto%253D%25252Flogin%25253Fto%25253D%2525252Flogin

I have also tried disabling Kaspersky Anti-virus (not relevant for the iPad) and switching off Lastpass password prompts (also not relevant for iPad).

**Comment by Mathiasw** [comment]
Attached are the console and network logs as bitmaps; the HAR file (whatever that is) would not upload.
This is Safari on Intel Sonoma, and I am running Seq in Docker Desktop 4.24.2.
If I deliberately give an incorrect password, I get a "bad password" banner, so it does seem to be contacting the licence server.

<img width="1865" alt="Seq Console Log 2023-11-08 at 06 03 36" src="https://github.com/datalust/seq-tickets/assets/733067/ee560458-7713-4402-9e2b-b417773dcbf7">
<img width="1883" alt="Seq Network Log 2023-11-08 at 06 01 27" src="https://github.com/datalust/seq-tickets/assets/733067/2a0619aa-52e1-42de-9358-9ad3d760f38b">

**Comment by Mathiasw** [comment]
Also, if I turn authentication off - install Seq on Docker Desktop without specifying a password hash - it works just fine.

**Reply by nblumhardt** [reply]
Thanks for all of the follow-up and info, @Mathiasw!

I'm not really sure where to look next - username/password authentication in Safari works normally for me, Sonoma on Intel 🤔 (We've also tried Safari on M1, we don't have an M2 handy but don't expect it to behave any differently.)

Hmm, actually, looking at your screenshot again - it appears that you're running an old version of the Seq Docker container.

`docker pull datalust/seq:latest` should get it updated. It'd be good to know whether this sorts it out!

Best regards,
Nick

**Comment by Mathiasw** [comment]
It seems to be fixed, on both my machines. 
I was using it with no userid or password (not a good idea I know) but at least I could access it in Safari.
Then I tried docker pull datalust/seq:latest and docker insisted the image was up to date, so I deleted the image and pulled it again. I didn't bother with this step on the second machine.
Then I set the admin account and userid from **within** seq, instead of passing the hash during docker installation, logged out and now I can also log in again!
So it seems to be fixed.

**Reply by nblumhardt** [reply]
That's great - thanks for the follow-up, @Mathiasw :+1:
